Formula/c/cucumber-cpp.rb

```diff
@@ -16,21 +17,25 @@ class CucumberCpp < Formula
   end
 
   depends_on "cmake" => :build
+  depends_on "nlohmann-json" => :build
+  # TODO: qt6 discovery not working, it finds 6.6.x but says it's unsuitable
+  depends_on "qt@5" => :build
   depends_on "ruby" => :test
+  depends_on "asio"
   depends_on "boost"
```

Comment:
`cucumber-cpp` removed need for Boost in 0.7.0: https://github.com/cucumber/cucumber-cpp/pull/275